content/academy/anti_scraping/techniques/rate_limiting.md

```diff
@@ -16,9 +16,40 @@ In cases when a higher number of requests is expected for the crawler, using a [
 
 ## [](#handling-rate-limiting) Dealing with sites which have rate-limiting
 
-In the Apify SDK, you can use [`browserPoolOptions.retireBrowserAfterPageCount`](https://github.com/apify/browser-pool#features) for browser-based crawlers (Puppeteer/Playwright) to retire a browser instance after a certain number of requests have been sent from it. In request-based crawlers, the [Session Pool](https://sdk.apify.com/docs/api/session-pool) can be leveraged.
-
-<!-- Add more here -->
+In the Apify SDK, you can use [`browserPoolOptions.retireBrowserAfterPageCount`](https://github.com/apify/browser-pool#features) for browser-based crawlers (Puppeteer/Playwright) to retire a browser instance after a certain number of requests have been sent from it. This will prevent the browser instance from sending too many requests to the page, avoiding the potential of hitting rate-limiting issues:
```

Comment:
This isn't true though. If you retire a browser, a different one will be spawned to fulfill your requests.

Rate-limiting depends on proxy rotation, not on using `retireBrowserAfterPageCount`. The option does not work on its own. The reason why it's used in this fashion is that you need to restart a browser to switch a proxy. So the real reason why the rate limit is overcome are the proxies. This needs to be explained well and the example needs to show proxy usage too.

It's quite complex, really. Especially if you factor in session pool. Because session pool does this retiring of browsers automatically for you, even if you don't use this option.

So I would vote for completely skipping this option and rethinking how this guide should be written from the ground up. There really aren't any meaningful shortcuts here.